exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggTemplate.java

```diff
@@ -573,6 +574,26 @@ public AggOutcome doWork() {
         delayedSetup();
       }
 
+      // If the prior call finished handling a prior incoming EMIT, need to call next() again
+      if ( !handleEmit && outcome == IterOutcome.EMIT ) {
```

Comment:
As discussed please try following:
1) Try to remove this code as its a duplicate and you will end up handling all the IterOutcomes again. Instead try setting `currentBatchRecordCount` to 0 once you are done processing all the incoming records. That way this code can be totally removed.
2) Please add few unit tests
3) Add an exception in case spilling happens in EMIT scenario.